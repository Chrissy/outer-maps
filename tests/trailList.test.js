import React from "react";
import TrailList from "../components/trailList.js";
import renderer from "react-test-renderer";

// the data piped into this is really complicated and probably a smell.

const trails = [
  {
    name: "fun trail",
    hasElevationData: true,
    id: 100,
    points: [
      {
        distanceFromPreviousPoint: 5
      },
      {
        distanceFromPreviousPoint: 2
      }
    ]
  },
  {
    name: "not fun trail",
    hasElevationData: false,
    id: 101,
    points: [
      {
        distanceFromPreviousPoint: 50
      },
      {
        distanceFromPreviousPoint: 1
      }
    ]
  },
  {
    name: "super fun trail",
    hasElevationData: true,
    id: 102,
    points: [
      {
        distanceFromPreviousPoint: 50000
      },
      {
        distanceFromPreviousPoint: 11
      }
    ]
  }
];

it("renders", () => {
  const tree = renderer
    .create(<TrailList unselectTrail={() => {}} trails={trails} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
