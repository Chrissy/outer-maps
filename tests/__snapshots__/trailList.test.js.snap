// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.emotion-15 {
  margin: 0.5em 0 3em 0;
  padding: 0 1em;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-radius: 0.5em;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #FF6000;
  box-sizing: border-box;
  color: #fff;
  font-size: 0.75em;
  margin-bottom: 1em;
  padding: 0.5em 0.75em;
}

.emotion-4:nth-of-type(3n + 2) {
  background-color: #CA4300;
}

.emotion-4:nth-of-type(3n + 3) {
  background-color: #9B3300;
}

.emotion-0 {
  padding: 0.5em 0.75em;
}

.emotion-1 {
  border-right: 1px solid #FF8244;
  padding: 0.5em 0.75em;
}

.emotion-2 {
  width: 2.5em;
  height: 0.75em;
  padding: 0 0.5em 0 0.25em;
  display: block;
}

.emotion-2 path {
  stroke: #fff;
}

<div
  className="emotion-15"
>
  <li
    className="emotion-4"
  >
    <div
      className="emotion-0"
    >
      fun trail
    </div>
    <div
      className="emotion-0"
    >
      <div
        className="emotion-1"
      >
        0m
      </div>
      <test-file-stub
        className="emotion-2"
        onClick={[Function]}
      />
    </div>
  </li>
  <li
    className="emotion-4"
  >
    <div
      className="emotion-0"
    >
      not fun trail
    </div>
    <div
      className="emotion-0"
    >
      <div
        className="emotion-1"
      >
        
      </div>
      <test-file-stub
        className="emotion-2"
        onClick={[Function]}
      />
    </div>
  </li>
  <li
    className="emotion-4"
  >
    <div
      className="emotion-0"
    >
      super fun trail
    </div>
    <div
      className="emotion-0"
    >
      <div
        className="emotion-1"
      >
        31.1m
      </div>
      <test-file-stub
        className="emotion-2"
        onClick={[Function]}
      />
    </div>
  </li>
</div>
`;
